Showing dialog for exception: addCharm() takes 3 positional arguments but 4 were given

## Report

Thank you for trying **conjure-up**! Before reporting a bug please make sure you've gone through this checklist:

- [ ] Is this problem already documented at https://docs.ubuntu.com/conjure-up/en/troubleshoot#common-spell-problems no
- [ ] Is conjure-up running inside a virtual machine? no
- [ ] Is this problem reproducible with `sudo snap refresh conjure-up --edge`? yes

## Please provide the output of the following commands

```
which juju
/snap/bin/juju

juju version
2.7-beta1-centos7-amd64

which conjure-up
/snap/bin/conjure-up

conjure-up --version
conjure-up 2.6.8


which lxc
/snap/bin/lxc

/snap/bin/lxc config show
config:
  core.https_address: '[::]'

/snap/bin/lxc version
Client version: 3.16
Server version: 3.16

cat /etc/lsb-release
cat: /etc/lsb-release: No such file or directory

```


Please attach tarball of **~/.cache/conjure-up**:

```
tar cvzf conjure-up.tar.gz ~/.cache/conjure-up
```

## Sosreport

Please attach a sosreport:

```
sudo apt install sosreport
sosreport
```

The resulting output file can be attached to this issue.


## What Spell was Selected? novalxd

## What provider (aws, maas, localhost, etc)? localhost

### MAAS Users
Which version of MAAS?

## Commands ran

Please outline what commands were run to install and execute conjure-up:
conjure-up
## Additional Information

2019-09-04 07:59:03,905 [INFO] conjure-up/openstack-novalxd - common.py:19 - Deploying Applications.
2019-09-04 07:59:04,030 [DEBUG] conjure-up/openstack-novalxd - events.py:52 - Setting Error at conjureup/events.py:149
2019-09-04 07:59:04,031 [ERROR] conjure-up/openstack-novalxd - events.py:161 - Unhandled exception in <Task finished coro=<do_deploy() done, defined at /snap/conjure-up/1055/lib/python3.6/site-packages/conjureup/controllers/juju/deploy/common.py:11> exception=TypeError('addCharm() takes 3 positional arguments but 4 were given',)>
Traceback (most recent call last):
  File "/snap/conjure-up/1055/lib/python3.6/site-packages/conjureup/controllers/juju/deploy/common.py", line 30, in do_deploy
    await app.juju.client.deploy(fn)
  File "/snap/conjure-up/1055/lib/python3.6/site-packages/juju/model.py", line 1308, in deploy
    await handler.execute_plan()
  File "/snap/conjure-up/1055/lib/python3.6/site-packages/juju/model.py", line 1991, in execute_plan
    result = await method(*step.args)
TypeError: addCharm() takes 3 positional arguments but 4 were given
2019-09-04 07:59:04,059 [DEBUG] conjure-up/openstack-novalxd - __init__.py:27 - Showing dialog for exception: addCharm() takes 3 positional arguments but 4 were given
2019-09-04 07:59:37,957 [DEBUG] conjure-up/openstack-novalxd - events.py:52 - Setting Shutdown at conjureup/events.py:145

--- a/Makefile
+++ b/Makefile
@@ -15,7 +15,7 @@ sysdeps:
 	    sudo add-apt-repository -y ppa:deadsnakes/ppa; \
 	fi
 	@sudo apt-get update
-	@sudo apt-get -qqyf install jq bsdtar bridge-utils software-properties-common snapcraft python3.7-dev tox shellcheck build-essential
+	@sudo apt-get -qqyf install jq bsdtar bridge-utils software-properties-common snapcraft python3-all-dev tox shellcheck build-essential
 
 travis-sysdeps:
 	@sudo add-apt-repository -y ppa:deadsnakes/ppa

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,6 +1,6 @@
 aiofiles==0.3.1
 Jinja2==2.10.1
-juju==0.11.6
+juju==2.6.3
 juju-wait==2.6.3
 prettytable==0.7.2
 progressbar2==3.20.0

--- a/snap/wrappers/placeholder
+++ b/snap/wrappers/placeholder
@@ -1,0 +1,2 @@
+#!/usr/bin/env bash
+# 'make test' will fail if this directory doesn't exist or doesn't contain any script.

